modules/deriving/src/test/scala/shapeless3/deriving/deriving.scala

```diff
@@ -78,6 +78,48 @@ class DerivationTests {
     assert(!v8.eqv(CCons(1, CCons(2, CCons(3, CNil))), CCons(1, CCons(4, CCons(3, CNil)))))
   }
 
+  @Test
+  def order: Unit = {
+    val v0 = Ord[Unit]
+    assert(v0.compare((), ()) == 0)
+
+    val v1 = Ord[Box[Int]]
+    assert(v1.compare(Box(1), Box(1)) == 0)
+    assert(v1.compare(Box(1), Box(0)) == 1)
+    assert(v1.compare(Box(0), Box(1)) == -1)
+
+    val v2 = Ord[OptionInt]
+    assert(v2.compare(NoneInt, NoneInt) == 0)
+    assert(v2.compare(SomeInt(0), SomeInt(0)) == 0)
+    assert(v2.compare(SomeInt(0), SomeInt(1)) == -1)
+    assert(v2.compare(SomeInt(1), SomeInt(0)) == 1)
+    assert(v2.compare(SomeInt(0), NoneInt) == -1)
```

Comment:
I guess I find this ordering a bit strange.

Reply:
It's definitely better than alphabetic ordering 😜

Reply:
:roll_eyes: